architxt/database/__init__.py

```diff
@@ -127,6 +134,35 @@ def read_table(
         yield Tree("ROOT", children)
 
 
+def read_unreferenced_table(parent_table: Table, foreign_table: ForeignKey, *, conn: Connection, sample: int = 0, visited_links: set[ForeignKey] = set()) -> Generator[Tree, None, None]:
+    """
+    Process the relations of a table that is not referenced by any other tables.
+
+    :param parent_table: The table that refers to the foreign table.
+    :param foreign_table: The table to process.
+    :param conn: SQLAlchemy connection.
+    :param sample: Number of samples for each table to get.
+    :param visited_links: Set of visited relations to avoid cycles.
+    :return: A list of trees representing the relations and data for the table.
+    """
+    table = foreign_table.column.table
+
+    query = table.select().outerjoin(
+        parent_table,
+        parent_table.c[foreign_table.parent.name] == table.c[foreign_table.column.name],
+        ).where(parent_table.c[foreign_table.parent.name] == None)
```

Comment:
You may whant to use `NOT EXISTS` here instead, something like

```suggestion
    query = table.select().where(
        ~exists()
        .where(foreign_table.parent == foreign_table.column)
    )
```